### ISSUE TITLE
[Feature Request] Expose token logprobs / logits during text generation in LiteRT-LM

### ISSUE THREAD

**Issue by alejrobinson** [google-ai-edge/LiteRT-LM#2540]
Currently, the LiteRT-LM run_decode()/send_message() API only returns the generated text. The underlying log probabilities computed during the forward pass are just discarded by the sampler (with the exception of run_text_scoring(), but that's not relevant here).

This prevents us from doing any kind of confidence scoring along with text generation. I want to be able to evaluate how sure the model is about each token of the answers it generates.

Could we get a flag like return_logprobs added to the config so it populates the token_scores array in the Responses object during text generation?

Thanks!
